Tests/TSCBasicTests/PathTests.swift

```diff
@@ -8,67 +8,67 @@
  See http://swift.org/CONTRIBUTORS.txt for Swift project authors
 */
 
-import XCTest
 import Foundation
-
 import TSCBasic
+import TSCTestSupport
+import XCTest
 
 class PathTests: XCTestCase {
 
     func testBasics() {
         XCTAssertEqual(AbsolutePath(path: "/").pathString, "/")
         XCTAssertEqual(AbsolutePath(path: "/a").pathString, "/a")
         XCTAssertEqual(AbsolutePath(path: "/a/b/c").pathString, "/a/b/c")
-        XCTAssertEqual(RelativePath(".").pathString, ".")
-        XCTAssertEqual(RelativePath("a").pathString, "a")
-        XCTAssertEqual(RelativePath("a/b/c").pathString, "a/b/c")
-        XCTAssertEqual(RelativePath("~").pathString, "~")  // `~` is not special
+        XCTAssertEqual(RelativePath(path: ".").pathString, ".")
+        XCTAssertEqual(RelativePath(path: "a").pathString, "a")
+        XCTAssertEqual(RelativePath(path: "a/b/c").pathString, "a/b/c")
+        //XCTAssertEqual(RelativePath(path: "~").pathString, "~")  // `~` is not special
```

Comment:
@neonichu @abertelrud had to comment this out since `~` does not validate as valid relative path. thoughts?

Reply:
Hm, why not? It's valid as a name, at least on macOS:

```
❯ mkdir ./~
❯ cd ./~
❯ pwd
/tmp/~
```

Reply:
Hmm... it actually _is_ are valid filename though (at least on Darwin and I suspect other environments with Unix-like path semantics).  `~` is a shell construct, and this test was specifically testing that.

Reply:
the point is that the validation code we have in place (regardless of this PR) does not agree :D so should we change that?

```
init(validatingRelativePath path: String) throws {
        switch path.first {
        case "/", "~":
            throw PathValidationError.invalidRelativePath(path)
        default:
            self.init(normalizingRelativePath: path)
        }
    }
```

Reply:
Yah, I think we should definitely change it, even disallowing `/` doesn't seem clear cut to me, but that at least doesn't prevent constructing certain valid paths.

Reply:
works for me. @abertelrud @elsh any concerns?